commit 613e9effd480add3cf5dc039ee9d947272ff735d
Author: Andreas Sturmlechner <andreas.sturmlechner@gmail.com>
Date:   Fri Sep 8 02:47:21 2017 +0200

    Revert "kio_pop3: Fix "missing mimetype" warnings."
    
    Unfortunately this breaks veryoldpim POP3, see also:
    https://git.reviewboard.kde.org/r/124987/
    
    This reverts commit 544410c905ce0ed2d40b33a50dcbb0796d6c91cd.

diff --git a/kioslave/pop3/pop3.cpp b/kioslave/pop3/pop3.cpp
index 8238a5fca..20730950d 100644
--- a/kioslave/pop3/pop3.cpp
+++ b/kioslave/pop3/pop3.cpp
@@ -812,7 +812,6 @@ void POP3Protocol::get(const KUrl & url)
        .
      */
     if (result) {
-      mimeType("text/plain");
       while (true /* !AtEOF() */ ) {
         memset(buf, 0, sizeof(buf));
         myReadLine(buf, sizeof(buf) - 1);
@@ -849,12 +848,6 @@ void POP3Protocol::get(const KUrl & url)
     m_cmd = CMD_NONE;
   } else if (cmd == "download" || cmd == "headers") {
     const QStringList waitingCommands = path.split(',', QString::SkipEmptyParts);
-    if ( waitingCommands.isEmpty() ) {
-      kDebug(7105) << "tried to request" << cmd << "for" << path << "with no specific item to get";
-      closeConnection();
-      error(ERR_INTERNAL, m_sServer);
-      return;
-    }
     bool noProgress = (metaData("progress") == "off"
                        || waitingCommands.count() > 1);
     int p_size = 0;
